web/tests/integration/app.test.ts

```diff
@@ -232,6 +233,127 @@ describe("user role", () => {
+  });
+
+  test("cannot reset client secret for another team", async () => {
+    const { rows: teams } = (await integrationDBExecuteQuery(
+      `SELECT id FROM "public"."team";`
+    )) as { rows: Array<{ id: string }> };
+
+    const { rows: teamMemberships } = (await integrationDBExecuteQuery(
+      `SELECT id, user_id, team_id FROM "public"."membership" WHERE "team_id" = '${teams[1].id}' AND "role" = 'OWNER' limit 1;`
+    )) as { rows: Array<{ id: string; user_id: string; team_id: string }> };
+
+    const { rows: teamApps } = (await integrationDBExecuteQuery(
+      `SELECT id FROM "public"."app" WHERE "team_id" = '${teams[0].id}' limit 1;`
+    )) as { rows: Array<{ id: string }> };
+
+    const tokenTeamId = teams[1].id;
+    const tokenUserId = teamMemberships[0].user_id;
+    const appId = teamApps[0].id;
+
+    const { req, res } = createMocks({
+      method: "POST",
+      headers: {
+        authorization: process.env.INTERNAL_ENDPOINTS_SECRET,
+      },
+      body: {
+        input: { app_id: appId },
+        action: { name: "reset_client_secret" },
+        session_variables: {
+          "x-hasura-role": "user",
+          "x-hasura-user-id": tokenUserId,
+          "x-hasura-team-id": tokenTeamId,
+        },
+      },
+    });
+
+    await handleSecretReset(req, res);
+
+    const responseData = res._getData();
+    const responseJSON = JSON.parse(responseData);
+    expect(responseJSON.extensions.code).toBe("insufficient_permissions");
+  });
+
+  test("can reset client secret", async () => {
+    const { rows: teams } = (await integrationDBExecuteQuery(
+      `SELECT id FROM "public"."team";`
+    )) as { rows: Array<{ id: string }> };
+
+    const { rows: teamMemberships } = (await integrationDBExecuteQuery(
+      `SELECT id, user_id, team_id FROM "public"."membership" WHERE "team_id" = '${teams[0].id}' AND "role" = 'OWNER' limit 1;`
+    )) as { rows: Array<{ id: string; user_id: string; team_id: string }> };
+
+    const { rows: teamApps } = (await integrationDBExecuteQuery(
+      `SELECT id FROM "public"."app" WHERE "team_id" = '${teams[0].id}' limit 1;`
+    )) as { rows: Array<{ id: string }> };
+
+    const tokenTeamId = teams[0].id;
+    const tokenUserId = teamMemberships[0].user_id;
+    const appId = teamApps[0].id;
+
+    const { req, res } = createMocks({
+      method: "POST",
+      headers: {
+        authorization: process.env.INTERNAL_ENDPOINTS_SECRET,
+      },
+      body: {
+        input: { app_id: appId },
+        action: { name: "reset_client_secret" },
+        session_variables: {
+          "x-hasura-role": "user",
+          "x-hasura-user-id": tokenUserId,
+          "x-hasura-team-id": tokenTeamId,
+        },
+      },
+    });
+
+    await handleSecretReset(req, res);
+
+    const responseData = res._getData();
+    const responseJSON = JSON.parse(responseData);
+    expect(responseJSON.client_secret).toBeDefined();
```

Comment:
@andy-t-wang generally a good practice to also check the DB to assert objects are in the state you expect